src/mapcraftercore/renderer/blockimages.cpp

```diff
@@ -109,6 +109,59 @@ bool DoubleChestTextures::load(const std::string& filename, int texture_size) {
 	return true;
 }
 
+bool ShulkerTextures::load(const std::string& base_filename, int texture_size) {
+    return (load_single(base_filename + "white.png", 0, texture_size) &&
+            load_single(base_filename + "orange.png", 1, texture_size) &&
+            load_single(base_filename + "magenta.png", 2, texture_size) &&
+            load_single(base_filename + "light_blue.png", 3, texture_size) &&
+            load_single(base_filename + "yellow.png", 4, texture_size) &&
+            load_single(base_filename + "lime.png", 5, texture_size) &&
+            load_single(base_filename + "pink.png", 6, texture_size) &&
+            load_single(base_filename + "gray.png", 7, texture_size) &&
+            load_single(base_filename + "silver.png", 8, texture_size) &&
+            load_single(base_filename + "cyan.png", 9, texture_size) &&
+            load_single(base_filename + "purple.png", 10, texture_size) &&
+            load_single(base_filename + "blue.png", 11, texture_size) &&
+            load_single(base_filename + "brown.png", 12, texture_size) &&
+            load_single(base_filename + "green.png", 13, texture_size) &&
+            load_single(base_filename + "red.png", 14, texture_size) &&
+            load_single(base_filename + "black.png", 15, texture_size));
+}
+
+bool ShulkerTextures::load_single(const std::string& filename, int color_index, int texture_size) {
+    RGBAImage image;
+    if (!image.readPNG(filename)) {
+        LOG(ERROR) << "Unable to read '" << filename << "'.";
+        return false;
+    }
+
+    if (image.getWidth() != image.getHeight()) {
+        LOG(ERROR) << "Shulker texture has invalid size (width:height must be 1:1): '"
+                   << filename << "'.";
+        return false;
+    }
+    // if the image is 64px wide, the shulker box images are 16x16
+    int ratio = image.getHeight() / 64;
```

Comment:
I think it still shows up as "changes requested", due to this not being an issue. I don't think I can flag it like that however 😛